x-pack/plugins/enterprise_search/server/routes/app_search/log_settings.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+import { IRouteDependencies } from '../../plugin';
+
+const logSettingsSchema = schema.object({
+  api: schema.maybe(
+    schema.object({
+      enabled: schema.boolean(),
+    })
+  ),
+  analytics: schema.maybe(
+    schema.object({
+      enabled: schema.boolean(),
+    })
+  ),
+});
+
+export function registerLogSettingsRoutes({
+  router,
+  enterpriseSearchRequestHandler,
+}: IRouteDependencies) {
+  router.get(
+    {
+      path: '/api/app_search/log_settings',
+      validate: false,
+    },
+    async (context, request, response) => {
+      return enterpriseSearchRequestHandler.createRequest({
+        path: '/as/log_settings',
+      })(context, request, response);
+    }
```

Comment:
Not a change request, just me wondering - do we want to simplify this to

```suggestion
    enterpriseSearchRequestHandler.createRequest({
      path: '/as/log_settings',
    })
```
at all? Or do we like the more explicit curried function notation that we end up having to use on some more complex API endpoints?

Reply:
Yeah think we should. It simplifies the code and tests a bit.

Reply:
c2528c01a5fa380c673d5df8fe4fa6bae61667de